src/main/java/capstone/mfslbackend/service/FantasyLeagueService.java

```diff
@@ -27,4 +30,19 @@ public List<FantasyLeague> getFantasyLeagueByName(String fantasyLeagueName) {
         String name = "%" + fantasyLeagueName + "%";
         return fantasyLeagueRepository.findFantasyLeagueByLeagueNameLikeIgnoreCase(name);
     }
-}
+
+    public FantasyLeague joinFantasyLeague(String userName, Long leagueId) {
+            Optional<FantasyLeague> fantasyLeagueOptional = getFantasyLeagueById(leagueId);
+            if (fantasyLeagueOptional.isEmpty()) {
+                log.error("Fantasy League with id {} not found", leagueId);
+                throw new IllegalArgumentException("Fantasy League with id " + leagueId + " not found");
+            }
+
+            FantasyLeague fantasyLeague = fantasyLeagueOptional.get();
+            User user = new User();
```

Comment:
Don't create a new user. You want to connect the existing user with that name. You have to use getUserById or something